### ISSUE TITLE
Big Sur 11.4 keyboard input monitoring

### ISSUE THREAD

**Issue by TommyCoyBoy** [OpenEmu/OpenEmu#4535]
Can't use OpenEmu as it can't monitor keyboard regardless of following troubleshooting steps.

# Steps to reproduce

1. Launch OpenEmu on BigSur 11.4
2. Open System Preferences from pop up "OpenEmu requires additional permissions"
3. Unlock System Preferences
4. Toggle off Input Monitoring for OpenEmu
5. Click Quit & Reopen popup
6. Close OpenEmu
7. Lock System Preferences and close it
8. Relaunch OpenEmu
9. Open System Preferences from pop up "OpenEmu requires additional permissions"
10. Unlock System Preferences
11. Toggle on Input Monitoring
12. Click Quit & Reopen popup

# Expected Behavior

1. Resolve bug of OpenEmu not monitoring keyboard input despite having permission.

# Observed Behavior

1. The popup asking for permission comes up no matter what I do to attempt to resolve it
2. The emulator launcher itself responds fine to keyboard inputs

# Debugging Information

I've attempted to resolve the issue with anything I can think of bar wiping OpenEmu or my computer itself and starting from a fresh install of both. I've restarted my computer and gone and closed anything I think could be interfering in the background via Activity Monitor.

- OpenEmu Version: 2.3.3
- macOS Version: Big Sur 11.4

**Comment by warrenson** [comment]
Same issue here. OSX 11.4 (Big Sur) + OpenEMU 2.3.3 6866.4-g70ac442b-Release

**Comment by HorrorSpirit** [comment]
I'm incredibly sorry, i'm new here and pretty stupid to have opened the same issue. My bad. I'm also getting this bug and waiting on a solution.

**Comment by ayevtushenko** [comment]
Any updates/plans to fix this issue?

**Comment by yolo2015** [comment]
I solved this problem.

Firstly I saw the openemu sign with a check box in the input monitoring session in the privacy.
Then there is a "+" and "-" sign. I selected openemu and click "-" sign to erase openemu.
After that I reopened Openemu, it showed that it requested input permission.
Go to privacy and see the openemu sign again in the input monitoring session.
Tick the check box and the problem is solved.

I do this accidentally and the result turns out is great.

Hope this can help. Enjoy gentleman.

**Comment by ayevtushenko** [comment]
> I solved this problem.
> 
> Firstly I saw the openemu sign with a check box in the input monitoring session in the privacy.
> Then there is a "+" and "-" sign. I selected openemu and click "-" sign to erase openemu.
> After that I reopened Openemu, it showed that it requested input permission.
> Go to privacy and see the openemu sign again in the input monitoring session.
> Tick the check box and the problem is solved.
> 
> I do this accidentally and the result turns out is great.
> 
> Hope this can help. Enjoy gentleman.

This does help me, thanks a lot man!

**Comment by TommyCoyBoy** [comment]
I've since been able to fix the issue.

I didn't even have to go as far as @yolo2015 did. All I had to was remove OpenEmu from the list of applications with Input Monitoring permissions under the Privacy tab in system preferences. I was not asked by OpenEmu to add it back with permissions after reopening it after this.

**Comment by StupidRepo** [comment]
Fixed this without doing what was said here:

> I solved this problem.
> 
> Firstly I saw the openemu sign with a check box in the input monitoring session in the privacy.
> Then there is a "+" and "-" sign. I selected openemu and click "-" sign to erase openemu.
> After that I reopened Openemu, it showed that it requested input permission.
> Go to privacy and see the openemu sign again in the input monitoring session.
> Tick the check box and the problem is solved.
> 
> I do this accidentally and the result turns out is great.
> 
> Hope this can help. Enjoy gentleman.

Just re-enable permissions (uncheck the box then check it) and click "Quit & Re-open" and it should sort it.

**Comment by ayevtushenko** [comment]
> Fixed this without doing what was said here:
> 
> > I solved this problem.
> > Firstly I saw the openemu sign with a check box in the input monitoring session in the privacy.
> > Then there is a "+" and "-" sign. I selected openemu and click "-" sign to erase openemu.
> > After that I reopened Openemu, it showed that it requested input permission.
> > Go to privacy and see the openemu sign again in the input monitoring session.
> > Tick the check box and the problem is solved.
> > I do this accidentally and the result turns out is great.
> > Hope this can help. Enjoy gentleman.
> 
> Just re-enable permissions (uncheck the box then check it) and click "Quit & Re-open" and it should sort it.

No, this didn’t work for me.

**Comment by deh6cker** [comment]
none of theese worked plz help

**Comment by alon-eth** [comment]
FYI - None of the above worked for me as well. Haven't been able to use it for a couple months now…￼￼

**Comment by nickve08** [comment]
Classic. Nothing was working even after removing and adding back in. Simple computer restart fixed it.

**Comment by clobber** [comment]
> Classic. Nothing was working even after removing and adding back in. Simple computer restart fixed it.

Sounds like possible application interference then. We note a restart may fix: https://github.com/OpenEmu/OpenEmu/wiki/Troubleshooting:-Input-problems#applications

**Comment by brpetrucci** [comment]
Tried running the find focus stealer thing on Stack Exchange and it prints OpenEmu, so I assume it's not an application interference problem. Restarting fixes but only temporarily. Have tried all other fixes here to no avail.

**Comment by iBulky** [comment]
I have the exact same problem, I am not able to change my controller nor am i able to move. I have already toggled and switched the input monitoring to on. I also close everything, redownloaded the app but it still doesnt work for me. Can anyone help me?

**Comment by schwannoma** [comment]
same problem here. Had been able to play without any problems on my Mac until I started scrambling with some terminal codes this morning. Tried all the methods mentioned above and still not working. Thought it'd probably be some deep level setting sh*t that involves more than just clicking on some buttons in the setting interface. :(

**Comment by schwannoma** [comment]
Damn I fixed the problem lol. The reason for my keyboard not responding to Openemu is the fact that I had been using iTerm2 and Mac terminals and kept it open ever since. The problem was solved the moment I turned off these two apps. 

PS. thanks to OpenEmu official documentation https://github.com/OpenEmu/OpenEmu/wiki/Troubleshooting:-Input-problems#applications in which iTerm2, Terminal and a few other apps are listed being capable of interfering with keyboard usage.
